### PR TITLE
Implements `ipinfo tool is_one_ip`

### PR DESCRIPTION
**Pull request by Harisabdullah**
Kindly ignore the mistake `isSingleIP` instead of `isOneIp`

### CODE REVIEW

File: ipinfo/cmd_tool.go [hunk 1]
```diff
@@ -38,6 +40,7 @@ Commands:
   prev         get the previous IP of the input IP
   is_v4        reports whether input is an IPv4 address.
   is_v6        reports whether input is an IPv6 address.
+  isOneIp      checks whether a CIDR contains exactly one IP.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool.go:43]:
```suggestion
  isOneIp      checks whether a CIDR or IP range contains exactly one IP.
```

File: ipinfo/cmd_tool_isOneIp.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		"--help": predict.Nothing,
+	},
+}
+
+func printHelpToolIsOneIp() {
+	fmt.Printf(
+		`Usage: %s tool isOneIp [<opts>] <cidr | filepath>
+
+Description:
+  checks whether a CIDR contains exactly one IP.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_isOneIp.go:24]: Same as above

File: ipinfo/cmd_tool_isOneIp.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+var completionsToolIsOneIp = &complete.Command{
+	Flags: map[string]complete.Predictor{
+		"-h":     predict.Nothing,
+		"--help": predict.Nothing,
+	},
+}
+
+func printHelpToolIsOneIp() {
+	fmt.Printf(
+		`Usage: %s tool isOneIp [<opts>] <cidr | filepath>
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_isOneIp.go:21]: Should also accept individual IP and IP ranges, just for completeness

File: lib/cmd_tool_isOneIp.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+func CmdToolIsOneIp(f CmdToolIsOneIpFlags, args []string, printHelp func()) error {
+	if len(args) == 0 || f.Help {
+		printHelp()
+		return nil
+	}
+
+	op := func(input string, inputType INPUT_TYPE) error {
+		switch inputType {
+		case INPUT_TYPE_CIDR:
+			fmt.Printf("%s %v\n", input, CIDRContainsExactlyOneIP(input))
```
Comment (UmanShahzad) [inline lib/cmd_tool_isOneIp.go:32]:
```suggestion
			fmt.Printf("%s,%v\n", input, CIDRContainsExactlyOneIP(input))
```

File: lib/cmd_tool_isOneIp.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+func CmdToolIsOneIp(f CmdToolIsOneIpFlags, args []string, printHelp func()) error {
+	if len(args) == 0 || f.Help {
+		printHelp()
+		return nil
+	}
+
+	op := func(input string, inputType INPUT_TYPE) error {
+		switch inputType {
+		case INPUT_TYPE_CIDR:
```
Comment (UmanShahzad) [inline lib/cmd_tool_isOneIp.go:31]: We should support single IPs (always true) and also IP ranges

**Review by UmanShahzad — COMMENTED** [review]
Needs to follow subcmd naming style as used in others (snake case not pascal / camel case).

Also see original linear issue to see intended spelling of the command.

File: ipinfo/cmd_tool.go [hunk 6]
```diff
@@ -38,6 +40,7 @@ Commands:
   prev         get the previous IP of the input IP
   is_v4        reports whether input is an IPv4 address.
   is_v6        reports whether input is an IPv6 address.
+  is_one_ip    checks whether a CIDR or ip range contains exactly one IP.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool.go:43]:
```suggestion
  is_one_ip    checks whether a CIDR or IP Range contains exactly one IP.
```

File: ipinfo/cmd_tool_isOneIp.go [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+		"--help": predict.Nothing,
+	},
+}
+
+func printHelpToolIsOneIp() {
+	fmt.Printf(
+		`Usage: %s tool is_one_ip [<opts>] <cidr | ip | ip-range | filepath>
+
+Description:
+  checks whether a CIDR or ip range contains exactly one IP.
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_isOneIp.go:24]: Checks whether a CIDR or IP Range contains exactly one IP.

**Review by UmanShahzad — COMMENTED** [review]
See https://github.com/ipinfo/cli/pull/168#issuecomment-1699560476

File: ipinfo/cmd_tool_isOneIp.go (hunk 7 above)
Comment (UmanShahzad) [inline ipinfo/cmd_tool_isOneIp.go:24]:
```suggestion
  Checks whether a CIDR or IP Range contains exactly one IP.
```

**Comment by UmanShahzad** [comment]
Lemme know when ready

File: lib/cmd_tool_isOneIp.go [hunk 8]
```diff
@@ -45,24 +49,6 @@ func CmdToolIsOneIp(f CmdToolIsOneIpFlags, args []string, printHelp func()) erro
...
-		return ipRange.Start == ipRange.End
-	} else if ipnet.IP.To16() != nil {
-		ipRange, _ := IP6RangeStrFromCIDR(cidrStr)
-		return ipRange.Start == ipRange.End
-	}
-
-	return false
-}
-
 func ipRangeContainsExactlyOneIP(ipRangeStr string) bool {
```
Comment (Harisabdullah) [inline lib/cmd_tool_isOneIp.go:52]: `parsePrefix` only parses from CIDR notation so `ipRangeContainsExactlyOneIP` had to be kept

**Review by UmanShahzad — APPROVED** [review]
Looks good but can we fix the filename to have `is_one_ip` instead of `isOneIp` too?

**Review by UmanShahzad — APPROVED** [review]

**Pull request merged**